crates/sdk/src/masp/utilities.rs

```diff
@@ -111,23 +105,31 @@ impl<C: Client + Send + Sync> LedgerMaspClient<C> {
                     .data
             };
 
-            for IndexedMaspData {
+            // Cache the last tx seen to avoid multiple deserializations
+            let mut last_tx: Option<(Tx, TxIndex)> = None;
+
+            for MaspEvent {
                 tx_index,
-                masp_refs,
+                kind: _,
```

Comment:
I cant tell where this field is used

Reply:
It is used in the event itself (which could be of type `masp/transfer` or `masp/fee-payment`), and after 837d480145d941d9b92f265b34d8ebd172bd7ccd it is also used when fetching txs from the indexer since when we cache them we need to be able to sort them correctly (as suggested by @sug0) and we need the tx kind for that since fee payment txs need to be move ahead of the queue